consumer.go

```diff
@@ -171,9 +221,19 @@ func (c *consumer) sync(force bool) (*ConsumerSync, error) {
 	}
 
 	// open remote reader
-	reader, err := NewReader(c.ctx, c.remote, &c.opt.ReaderOptions)
-	if err != nil {
-		return nil, err
+	var reader *Reader
+	if c.opt.incremental {
+		var remotes []*bfs.Object
+		if reader, remotes, err = c.newIncrementalReader(); err != nil {
+			return nil, err
+		}
+		for _, r := range remotes {
+			defer r.Close()
```

Comment:
never defer in a loop, it's not what you want. the `defer reader.Close()` below should take care of everything